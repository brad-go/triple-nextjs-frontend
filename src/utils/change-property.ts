export const changePropertyCamelToKebab = (property: string) => {
  const capital = property.match(/[A-Z]/g)

  if (capital) {
    const capitalIndex = property.indexOf(capital[0])
    const firstWord = property.slice(0, capitalIndex)
    const secondWord = capital[0]
      .toLocaleLowerCase()
      .concat(property.slice(capitalIndex + 1))

    property = `${firstWord}-${secondWord}`
  }

  return property
}
