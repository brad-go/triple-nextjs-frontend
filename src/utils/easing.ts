export function easeOutExpo(
  time: number,
  start: number,
  end: number,
  duration: number,
) {
  if (time === duration) {
    return start + end
  }

  return end * (-Math.pow(2, (-10 * time) / duration) + 1) + start
}
