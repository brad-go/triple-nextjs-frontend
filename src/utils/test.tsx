import { PropsWithChildren, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from 'styles/theme'

function Provider<T>({ children }: PropsWithChildren<T>) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Provider, ...options })

export { customRender as render }
