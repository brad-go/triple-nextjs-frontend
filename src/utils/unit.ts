export const setUnit = (value: number | string, unit = 'px') => {
  if (typeof value === 'string') {
    return value
  }

  return `${value}${unit}`
}
