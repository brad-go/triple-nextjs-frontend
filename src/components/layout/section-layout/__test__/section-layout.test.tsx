import { screen } from '@testing-library/dom'

import SectionLayout from '../section-layout'

import { render } from 'utils/test'

describe('SectionLayout', () => {
  it('should be rendered', () => {
    const { container } = render(<SectionLayout />)

    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should be rendered with children', () => {
    const { getByText } = render(
      <SectionLayout>
        <h2>Example</h2>
      </SectionLayout>,
    )

    expect(getByText('Example')).toBeInTheDocument()
  })

  it('should be rendered with different style', () => {
    render(
      <SectionLayout backgroundStyle="primary" sectionType="primary">
        <h2>Primary</h2>
      </SectionLayout>,
    )

    expect(screen.getByText('Primary')).toBeInTheDocument()

    render(
      <SectionLayout backgroundStyle="secondary" sectionType="primary">
        <h2>Secondary</h2>
      </SectionLayout>,
    )

    expect(screen.getByText('Secondary')).toBeInTheDocument()

    render(
      <SectionLayout backgroundStyle="toolbox" sectionType="primary">
        <h2>Toolbox</h2>
      </SectionLayout>,
    )

    expect(screen.getByText('Toolbox')).toBeInTheDocument()

    render(
      <SectionLayout backgroundStyle="recruit" sectionType="primary">
        <h2>Recruit</h2>
      </SectionLayout>,
    )

    expect(screen.getByText('Recruit')).toBeInTheDocument()

    render(
      <SectionLayout backgroundStyle="footer" sectionType="footer">
        <h2>Footer</h2>
      </SectionLayout>,
    )

    expect(screen.getByText('Footer')).toBeInTheDocument()
  })
})
