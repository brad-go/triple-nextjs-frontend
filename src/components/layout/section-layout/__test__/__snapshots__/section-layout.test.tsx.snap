// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionLayout should be rendered 1`] = `
.c0 {
  position: relative;
  background-position: center center;
  background-size: cover;
  overflow: hidden;
  background-image: inherit;
  background-repeat: inherit;
  background-attachment: inherit;
  background-origin: inherit;
  background-clip: inherit;
  background-color: inherit;
}

.c1 {
  position: relative;
  margin: 0px auto;
}

@media screen and (min-width:1142px) {
  .c1 {
    width: 1040px;
    height: auto;
  }
}

@media screen and (max-width:1141px) {
  .c1 {
    width: 360px;
    height: auto;
  }
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    />
  </section>
</div>
`;
